src/box/memtx_engine.cc

```diff
@@ -614,7 +614,9 @@ memtx_engine_prepare(struct engine *engine, struct txn *txn)
 	if (memtx_tx_manager_use_mvcc_engine) {
 		struct txn_stmt *stmt;
 		stailq_foreach_entry(stmt, &txn->stmts, next) {
-			assert(stmt->space->engine == engine);
+			assert(stmt->space->engine == stmt->engine);
+			if (stmt->engine != engine)
+				continue;
```

Comment:
If memtx mvcc is enabled, there won't be any alien statements on the list, no?

Reply:
Yes, this change is unnecessary at this point.